spec/consensus/time.md

```diff
@@ -0,0 +1,52 @@
+---
+order: 2
+---
+# Time
+
+CometBFT provides a Byzantine fault-tolerant source of time.
+
+Time in CometBFT is defined with the [`Time`][spec-time] field of the
+block [`Header`][spec-header].
+
+## Properties
+
+The Time produced by CometBFT satisfies the following properties:
+
+- **Time Monotonicity**: time is monotonically increasing.  More precisely, given
+  two block headers `H1` of height `h1` and `H2` of height `h2`,
+  it is guaranteed that if `h2 > h1` then `H2.Time > H1.Time`.
+
+- **Byzantine Fault Tolerance**: faulty or malicious processes should not be able
+  to arbitrarily influence (increase or decrease) the block Time.
+  In other words, the Time of blocks should be defined by correct processes.
+
+In addition, the Time produced by CometBFT is expected, by external observers, to provide:
+
+- **Relation to real time**: block times bear some resemblance to real time.
+  In other words, block times should represent, within some reasonable accuracy,
+  the actual time at which blocks were produced.
```

Comment:
This definition is problematic but I have no better one to offer.

Reply:
Indeed...

Reply:
An idea would be to express it with a bound, unknown but existing. Something like:

_ block times bear some resemblance to real time. For every decided block `H` of height `h`, there exists an unknown bound `B` such that, `| H.Timestamp - real_time | < B`, where `real_time` is the actual time when `H` was first proposed.

Anyway, I'm not sure about this. You may well decide to keep the current text, I'm OK with that.

Reply:
We had an `ACCURACY` parameter in the original spec. But relating it to `PRECISION` is not really trivial, although one can assume that `ACCURACY <= 2*PRECISION`.

In this case, we have to add a `2` factor for all `PRECISION` entries in the timely properties. Which has rendered the spec more complex, and we are not sure if this reasoning is right.

Reply:
See: https://github.com/cometbft/cometbft/blob/master/spec/consensus/proposer-based-timestamp/pbts-sysmodel_002_draft.md#accuracy

Reply:
My suggestion is really basic... definitely not proposing to change the sysmodel in the spec. It was just another possible wording for the "Relation to real time" definition.